From ba74d639178916221c748b0d5d89f7ac4f5ed669 Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Sat, 22 Apr 2017 14:00:33 +0200
Subject: [PATCH] Revert "Disallow executing Dolphin as root on Linux"

This reverts commit 0bdd8e0b0516555c6233fdc7901e9b417cf89791.
We ship a desktop file to open dolphin as root and we allow YaST on the
desktop. So this patch is absolutely pointless for us.
---
 src/main.cpp | 11 -------------
 1 file changed, 11 deletions(-)

Index: dolphin-19.11.70git.20230131T022613~e4e68b90b/src/main.cpp
===================================================================
--- dolphin-19.11.70git.20230131T022613~e4e68b90b.orig/src/main.cpp	2023-01-31 03:26:13.000000000 +0100
+++ dolphin-19.11.70git.20230131T022613~e4e68b90b/src/main.cpp	2023-01-31 07:40:48.073527099 +0100
@@ -43,25 +43,8 @@
 #endif
 #endif
 
-#ifndef Q_OS_WIN
-#include <unistd.h>
-#endif
-#include <iostream>
-
 int main(int argc, char **argv)
 {
-#ifndef Q_OS_WIN
-    // Prohibit using sudo or kdesu (but allow using the root user directly)
-    if (getuid() == 0) {
-        if (!qEnvironmentVariableIsEmpty("SUDO_USER")) {
-            std::cout << "Running Dolphin with sudo can cause bugs and expose you to security vulnerabilities." << std::endl;
-            return EXIT_FAILURE;
-        } else if (!qEnvironmentVariableIsEmpty("KDESU_USER")) {
-            std::cout << "Running Dolphin with kdesu can cause bugs and expose you to security vulnerabilities." << std::endl;
-            return EXIT_FAILURE;
-        }
-    }
-#endif
 
     /**
      * enable high dpi support
